pkg/kubelet/server/server.go

```diff
@@ -729,9 +731,36 @@ func (s *Server) getContainerLogs(request *restful.Request, response *restful.Re
 		response.WriteError(http.StatusInternalServerError, fmt.Errorf("unable to convert %v into http.Flusher, cannot show logs", reflect.TypeOf(response)))
 		return
 	}
-	fw := flushwriter.Wrap(response.ResponseWriter)
+
+	var (
+		stdout io.Writer
+		stderr io.Writer
+		fw     = flushwriter.Wrap(response.ResponseWriter)
+	)
+	if utilfeature.DefaultFeatureGate.Enabled(features.SplitStdoutAndStderr) {
+		wantedStream := logOptions.Stream
+		// No stream type specified, default to All
+		if wantedStream == nil {
+			allStream := v1.LogStreamTypeAll
+			wantedStream = &allStream
+		}
+		switch *wantedStream {
+		case v1.LogStreamTypeStdout:
+			stdout, stderr = fw, nil
+		case v1.LogStreamTypeStderr:
+			stdout, stderr = nil, fw
+		case v1.LogStreamTypeAll:
+			stdout, stderr = fw, fw
+		default:
+			_ = response.WriteError(http.StatusBadRequest, fmt.Errorf("invalid stream type %q", *logOptions.Stream))
+			return
+		}
+	} else {
+		stdout, stderr = fw, fw
```

Comment:
if feature gate is NOT set, but `wantedStream` is set to anything except `all`, we should return an error.

Reply:
done